src/ast.c

```diff
@@ -109,14 +110,19 @@ static jl_ast_context_t jl_ast_main_ctx;
 #define jl_ast_context_list_item(node)          \
     container_of(node, jl_ast_context_t, list)
 
-#define JL_AST_PRESERVE_PUSH(ctx, _roots, old_roots)    \
-    jl_array_t *_roots = NULL;                          \
-    jl_array_t **old_roots = ctx->roots;                \
-    ctx->roots = &_roots;                               \
-    JL_GC_PUSH1(&_roots)
-#define JL_AST_PRESERVE_POP(ctx, old_roots)     \
+#define JL_AST_PRESERVE_PUSH(ctx, _roots, old, inmodule)  \
+    jl_array_t *(_roots) = NULL;                \
+    struct { jl_array_t **roots;                \
+             jl_module_t *module; } (old);      \
```

Comment:
I suspect the later errors are nonsense after the first syntax error. What would it look like if given a name?